# In case of an unknown/invalid encoding, id3_parse_string() will
# return NULL, but the return value wasn't checked resulting
# in segfault in id3_ucs4_length().  This is the only place
# the return value wasn't checked.
# This patch was taken from debian patches for libid3tag
--- libid3tag-0.15.1b~/src/compat.gperf	2004-01-23 09:41:32.000000000 +0000
+++ libid3tag-0.15.1b/src/compat.gperf	2007-01-14 14:36:53.000000000 +0000
@@ -236,6 +236,10 @@
 
     encoding = id3_parse_uint(&data, 1);
     string   = id3_parse_string(&data, end - data, encoding, 0);
+    if (!string)
+    {
+	continue;
+    }
 
     if (id3_ucs4_length(string) < 4) {
       free(string);
--- libid3tag-0.15.1b~/src/parse.c	2004-01-23 09:41:32.000000000 +0000
+++ libid3tag-0.15.1b/src/parse.c	2007-01-14 14:37:34.000000000 +0000
@@ -165,6 +165,9 @@
   case ID3_FIELD_TEXTENCODING_UTF_8:
     ucs4 = id3_utf8_deserialize(ptr, length);
     break;
+  default:
+  	/* FIXME: Unknown encoding! Print warning? */
+	return NULL;
   }
 
   if (ucs4 && !full) {
